### ISSUE TITLE
API - unauthenticated join returns a HTTP status 302 redirect, no XML

### ISSUE THREAD

**Issue by NPCGundlach** [bigbluebutton/bigbluebutton#9749]
**Describe the bug**
When you call the API to join a user into a meeting but use a wrong or missing password the answer  is a HTTP status 302, the given URL is 
`<server-URL>/?errors=[message:<errorMessage>...key: <errorKey>]`, the answer to that contains a HTTP status 303/307, redirecting to `<server-URL>`.

**To Reproduce**
1. Go to https://mconf.github.io/api-mate/
2. Create a new meeting with the default values by clicking on the "GET" button
3. Empty the content of the "attendeePW" field
4. Click on the "GET" button of the "join as attendee" request
5. Observe the site that is being loaded.

**Expected behavior**
If the join endpoint receives invalid credentials, it should answer with an XML document as described in the [documentation about error handling](http://docs.bigbluebutton.org/dev/api.html#error-handling).

**Actual behavior**
The endpoint answers with a status 302, redirecting to an url with the error key and message contained in the query parameter `errors`, which further redirects to the root of the current (sub-)domain via a status 303/307.  
I've seen the 307 on our system here, the system API-Mate uses per default returns 303.

**Answer Headers**
```
HTTP/2 302 Found
server: nginx/1.10.3 (Ubuntu)
date: Wed, 03 Jun 2020 07:42:28 GMT
content-length: 0
x-application-context: application:production:8090
location: https://m044.rna1.blindsidenetworks.com?errors=[%7B%22message%22:%22You%20either%20did%20not%20supply%20a%20password%20or%20the%20password%20supplied%20is%20neither%20the%20attendee%20or%20moderator%20password%20for%20this%20conference.%22,%22key%22:%22invalidPassword%22%7D]
p3p: CP="No P3P policy available"
X-Firefox-Spdy: h2
```

```
HTTP/2 303 See Other
server: nginx/1.10.3 (Ubuntu)
date: Wed, 03 Jun 2020 07:42:28 GMT
content-type: text/html
content-length: 178
location: https://blindsidenetworks.com/
X-Firefox-Spdy: h2
```

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Firefox 76

**Comment by ib888877** [comment]
I'm facing the same error, anyone can help

**Comment by matiasilva** [comment]
@ib888877 what error are you facing? Supply the correct attendee password to your API call

**Comment by ib888877** [comment]
I found it, when I set guestPolicy to ALWAYS_DENY the meeting is not accepting to join as attendee

**Comment by NPCGundlach** [comment]
@matiasilva 
The issue is more that this behavior is hard to work with if you want to use the API. Almost every other API response comes as a XML response conforming to

```xml
<response>
  <success>successState</success>
  <messagekey>reasonKey</messagekey>
  <message>reason</message>
  ...
</response>
```

which can be parsed, logged and reacted to relatively easy.  
`/join` instead giving out a HTTP Status 302 response with the error message in the URL query parameters in the Location header of the response is unexpected and, as far as I can see, undocumented in the [API documentation](https://docs.bigbluebutton.org/dev/api.html#error-handling).  

The core of the issue is not "I provided invalid information, please help", it is "Either the application has a bug or the documentation is outdated, either should be fixed".  

The documentation recommendation about "join" calls is 
> You should simply redirect the user to the call URL, and they will be entered into the meeting.

[Source](https://docs.bigbluebutton.org/dev/api.html#join)
If we were doing that, users encountering this error would just be dropped on the domain root, without any way for us to catch them. The only ways to be aware of this happening would be to either monitor all HTTP requests and looking at any that contain the "error" query parameter, or by monitoring the log file for that specific error, as it is categorized under "DEBUG" and not "ERROR" - provided the developer in question even knows about this, as, like I wrote before, this behavior is not documented in the API documention.

This also seems to apply to at least one other error - e.g. a join request with a meetingID that hasn't yet been created causing a invalidMeetingIdentifier error also seems to redirect on a 302->307 on our system - although in the apiMate that instead seems to result in a NotFound error with an XML response. Can't see any relevant fix in the release notes between 2.3.2 (which our test server uses) and the current 2.3.6 that would indicate that the invalidMeetingIdentifier error has been removed, so not sure what's up with that.

Easiest way to resolve this issue would be to include in the API documentation of "join" calls that they can occasionally respond with a HTTP status 302, containing the actual error in the Location header of the response. Although personally I would prefer it if no errors would do that and the error would result in an XML response instead, or in a proper error page if the "join" api call is opened without the redirect parameter.
